### PR TITLE
[READY] Implement Sortable where possible

### PR DESCRIPTION
**Pull request by tgalopin**
As proposed in #8, I'm implementing a Sortable system.

Agenda:

- Rename `CachedStore` in `CachingDecorator` ✔
- Create a `SortableStore` interface for sortable stores ✔
- Create a `SortableDecorator` with a `sort()` method that use an array for an in-memory sort for others ✔
- Implement `sort()` for native sortable stores **WIP**

I'm currently wondering if I should create a `Decorator` namespace and a `Decorator` interface (as with this PR and the PR about countable, we will have 3 decorators). What do you think?

### CODE REVIEW

File: src/CachingDecorator.php [hunk 1]
```diff
@@ -45,6 +45,8 @@ class CachedStore implements KeyValueStore
      * @param Cache         $cache The cache.
      * @param int           $ttl   The time-to-live for cache entries. If set to
      *                             0, cache entries never expire.
+     *
+     * @throws \InvalidArgumentException If the provided cache is not supported
```
Comment (webmozart) [inline src/CachingDecorator.php]: You don't need to include the leading `\` since you imported the class.

**Comment by webmozart** [comment]
Thanks for working on this @tgalopin! Looks good :) Sure, you can add a Decorator namespace if you like.

**Comment by tgalopin** [comment]
I think I'll stay like that after all. Decorators are just specific stores (they implement the same API) so it's not that incoherent.

**Comment by tgalopin** [comment]
I think it's ready as I'm going to fix the Scrunitizer problems in another PR (#11). What do you think @webmozart ?

File: src/Api/KeyValueStore.php [hunk 2]
```diff
@@ -52,6 +52,8 @@
      * @param int|string $key   The key to set.
      * @param mixed      $value The value to set for the key.
      *
+     * @return void
```
Comment (webmozart) [inline src/Api/KeyValueStore.php]: Can you remove this and the next definition?

**Comment by webmozart** [comment]
Could you rebase, squash, `push -f` and fix the StyleCI issues please?

File: src/Api/KeyValueStore.php (hunk 2 above)
Comment (tgalopin) [inline src/Api/KeyValueStore.php]: Actually, it's something required by ScrunitizerCI. Should I remove it anyway?

Comment (webmozart) [inline src/Api/KeyValueStore.php]: Yes, please. I think I disabled this rule now (you never know..) :)

**Comment by tgalopin** [comment]
I'm not sure of my squash, is it fine for you?

**Comment by webmozart** [comment]
-> IRC

**Comment by webmozart** [comment]
Could you add tests for the SortableDecorator please?

**Comment by tgalopin** [comment]
We still have the `@return void` problem in Scrunitizer but other than that, it seems fine :) .

**Comment by webmozart** [comment]
Thanks @tgalopin! :)

**Pull request merged**